File: Q001-050/Q020.R
# encoding=utf-8

f <- function(n) {
	a <- sprintf('%.0f', prod(1:n))
	b <- strsplit(a, '')[[1]]
	c <- sum(as.integer(b))
	cat(sprintf('Sum of the digits in the number %d! = %d\n', n, c))
	c
}

### Test:
f(10)
f(100)

require(Rmpfr)
f.mpfr <- function(n) {
	v <- factorialMpfr(n)
	vs <- as(v, 'character')
	vc <- strsplit(vs, '')[[1]]
	sum(as.integer(vc))
}

f.mpfr(10)
f.mpfr(100)

File: Q001-050/Q020.py
#!/usr/bin/env python
# encoding=utf-8

import math

def digitSumOfFactorial(n):
	return sum([int(x) for x in str(math.factorial(n))])

if __name__=='__main__':
	print 'Sum of digits of 10! =', digitSumOfFactorial(10)
	print 'Sum of digits of 100! =', digitSumOfFactorial(100)

